libs/api_common/api_common.cpp

```diff
@@ -49,3 +52,48 @@ get_file_size(const char* filename, size_t* byte_code_size)
 
     return result;
 }
+
+ebpf_result_t
+windows_error_to_ebpf_result(uint32_t error)
+{
+    switch (error) {
+    case ERROR_SUCCESS:
+        return EBPF_SUCCESS;
+
+    case ERROR_INVALID_HANDLE:
+        return EBPF_ERROR_INVALID_HANDLE;
+
+    case ERROR_FILE_NOT_FOUND:
+        return EBPF_FILE_NOT_FOUND;
+
+    case ERROR_NOT_ENOUGH_MEMORY:
+        return EBPF_NO_MEMORY;
+    }
+
+    return EBPF_FAILED;
+}
+
+uint32_t
```

Comment:
can this return ebpf_result_t?  (even if it means you have to call windows_error_to_ebpf_result before returning)